### ISSUE TITLE
youtube-dl -U doesn't work anymore

### ISSUE THREAD

**Issue by leooleo-habbib** [rg3/youtube-dl#4666]
Hi, I used to update youtube-dl regulary using the '-U' which worked all the time. But now:  

Updating to version 2015.01.08 ...
ERROR: unable to download latest version
jachmann@hurx:~/src/youtube-dl>sudo youtube-dl -Uv
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-Uv']
[debug] Encodings: locale ISO8859-15, fs ISO8859-15, out ISO8859-15, pref ISO8859-15
[debug] youtube-dl version 2015.01.07.2
[debug] Python version 2.7.9 - FreeBSD-10.1-STABLE-i386-32bit-ELF
[debug] exe versions: ffmpeg N-45446-g037b527, ffprobe N-44582-g4e9745f, rtmpdump 2.4
[debug] Proxy map: {}
Updating to version 2015.01.08 ...
Traceback (most recent call last):
  File "/usr/local/bin/youtube-dl/youtube_dl/update.py", line 169, in update_self
    urlh = compat_urllib_request.urlopen(version['bin'][0])
  File "/usr/local/lib/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/lib/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "/usr/local/lib/python2.7/urllib2.py", line 449, in _open
    '_open', req)
  File "/usr/local/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python2.7/urllib2.py", line 1240, in https_open
    context=self._context)
  File "/usr/local/lib/python2.7/urllib2.py", line 1197, in do_open
    raise URLError(err)
URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:581)>

ERROR: unable to download latest version

Maybe something with my Pythen Installatione ? 

Hope that helps.
Thanks
Christian Jachmann

**Comment by phihag** [comment]
Most likely the result of upgrading to Python 2.7.9, which does include certificate verification, and a botched certificate setup on FreeBSD.

Just for reference, can you visit https://yt-dl.org/ in a browser?
What does `curl https://yt-dl.org/ >/dev/null` or `wget -O /dev/null https://yt-dl.org/` output?

**Comment by leooleo-habbib** [comment]
Thu, Jan 08, 2015 at 09:23:18AM -0800, Philipp Hagemeister :
> Most likely the result of upgrading to Python 2.7.9, which does include certificate verification, and a botched certificate setup on FreeBSD.
> 
> Just for reference, can you visit https://yt-dl.org/ in a browser?
> What does `curl https://yt-dl.org/ >/dev/null` or `wget -O /dev/null https://yt-dl.org/` output?
> 

So here are:

jachmann@hurx:~>curl https://yt-dl.org/ >/dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 28207    0 28207    0     0   139k      0 --:--:-- --:--:-- --:--:--  139k


jachmann@hurx:~>wget -O /dev/null https://yt-dl.org/
converted 'https://yt-dl.org/' (ISO8859-15) -> 'https://yt-dl.org/' (UTF-8)
--2015-01-09 01:07:08--  https://yt-dl.org/
Auflösen des Hostnamen »yt-dl.org (yt-dl.org)«... 95.143.172.170, 2001:1a50:11:0:5f:8f:acaa:177
Verbindungsaufbau zu yt-dl.org (yt-dl.org)|95.143.172.170|:443... verbunden.
HTTP-Anforderung gesendet, warte auf Antwort... 200 OK
Länge: nicht spezifiziert [text/html]
In »»/dev/null«« speichern.

/dev/null               [ <=>                  ]  27,55K  --.-KB/s   in 0,06s  

2015-01-09 01:07:09 (473 KB/s) - »/dev/null« gespeichert [28207]

jachmann@hurx:~>

So, don't know what this mean,a

But:

jachmann@hurx:~>sudo youtube-dl -U
Passwort:
Updating to version 2015.01.08 ...
ERROR: unable to download latest version
jachmann@hurx:~>

somth. w. SSL, or so ?

**Comment by phihag** [comment]
Thank you for the report. I believe this issue to be fixed in youtube-dl 2015.01.09 and newer. Unfortunately, to get there, you'll have to run a workaround command once:
```
sudo wget https://yt-dl.org/latest/youtube-dl -O /usr/local/bin/youtube-dl
sudo chmod a+x /usr/local/bin/youtube-dl
```
After that, updating should work again.

**Comment by leooleo-habbib** [comment]
Fri, Jan 09, 2015 at 11:23:38AM -0800, Philipp Hagemeister :
> Thank you for the report. I believe this issue to be fixed in youtube-dl 2015.01.09 and newer. Unfortunately, to get there, you'll have to run a workaround command once:
> ```
> sudo wget https://yt-dl.org/latest/youtube-dl -O /usr/local/bin/youtube-dl
> sudo chmod a+x /usr/local/bin/youtube-dl
> ```
> After that, updating should work again.
> 

Good Work, Thank You

Christian Jachmann
